engine/OthelloEngine.py

```diff
@@ -48,9 +51,57 @@ def play_game(self):
       # remember to add each move to all_moves
       # Also track individual turn times as well as total time
       # Return the winner: 'W', 'B', or 'T'
-      pass
-      
-   
+
+      # Sanity check for self.check_end()
+      while '-' in row in self.game_state:
+         # Each team takes their turn
+         for team in (self.white_team, self.black_team):
+            move = self.record_turn(team)
+
+            # execute_turn returns a character IFF the team's move
+            # causes them to lose automatically
+            if type(move) != tuple:
+               return move
+
+            # Else update the board and check if the game is over
+            else:
+               self.update_board(move)
+               self.all_moves.append(move)
+
+               gameEnd = self.check_end();
+               if gameEnd != None:
+                  return gameEnd
+
+         # Increment turn
+         self.turn_number += 1
```

Comment:
This should be indented (each turn is a different number, even turn numbers being black, and odd numbers being white)